dev_docs/tutorials/building_a_plugin.mdx

```diff
@@ -42,14 +42,33 @@ plugins/
   "id": "demo",
   "version": "kibana",
   "server": true,
-  "ui": true
+  "ui": true,
+  "owner": { [1]
+    "name": "App Services",
+    "githubTeam": "kibana-app-services"
+  },
+  "description": "This plugin extends Kibana by doing xyz, and allows other plugins to extend Kibana by offering abc functionality. It also exposes some helper utilities that do efg", [2]
+  "requiredPlugins": ["data"], [3]
+  "optionalPlugins": ["alerting"] [4]
+  "requiredBundles": ["anotherPlugin"] [5]
 }
 ```
 
+[1], [2]: Every internal plugin should fill in the owner and description properties.
+
+[3], [4]: Any plugin that you have a dependency on should be listed in `requiredPlugins` or `optionalPlugins`. Doing this will ensure that you have access to that plugin's start and setup contract inside your own plugin's start and setup lifecycle methods. If a plugin you optionally depend on is not installed, it will be undefined if you try to access it. If a plugin you require is not installed, kibana will fail to build.
```

Comment:
```suggestion
[3], [4]: Any plugin that you have a dependency on should be listed in `requiredPlugins` or `optionalPlugins`. Doing this will ensure that you have access to that plugin's start and setup contract inside your own plugin's start and setup lifecycle methods. If a plugin you optionally depend on is not installed or disabled, it will be undefined if you try to access it. If a plugin you require is not installed or disabled, kibana will fail to build.
```